tests/components/broadlink/test_config_flow.py

```diff
@@ -0,0 +1,465 @@
+"""Test the Broadlink config flow."""
+import errno
+import socket
+
+import broadlink.exceptions as blke
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.broadlink.const import DOMAIN
+
+from . import pick_device
+
+from tests.async_mock import MagicMock, call, patch
+
+
+@pytest.fixture(autouse=True)
+def broadlink_module_fixture():
+    """Mock broadlink module."""
+    broadlink = MagicMock()
+    with patch.dict("sys.modules", {"broadlink": broadlink}):
+        yield broadlink
+
+
+@pytest.fixture(autouse=True)
+def broadlink_setup_fixture():
+    """Mock broadlink entry setup."""
+    with patch(
+        "homeassistant.components.broadlink.async_setup_entry", return_value=True
+    ):
+        yield
+
+
+async def test_flow_user_works(hass):
+    """Test a config flow initiated by the user.
+
+    Best case scenario with no errors or locks.
+    """
+    device = pick_device(0)
+    mock_device = device.get_mock_api()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    with patch("broadlink.discover", return_value=[mock_device]) as mock_discover:
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host, "timeout": device.timeout},
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "finish"
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {"name": device.name},
+    )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == device.name
+    assert result["data"] == device.get_entry_data()
+
+    assert mock_discover.call_count == 1
+    assert mock_device.auth.call_count == 1
+
+
+async def test_flow_user_already_in_progress(hass):
+    """Test we do not accept more than one config flow per device."""
+    device = pick_device(0)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("broadlink.discover", return_value=[device.get_mock_api()]):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host, "timeout": device.timeout},
+        )
+
+    result = await hass.config_entries.flow.async_init(
```

Comment:
When can this happen?

Reply:
1. Open two tabs in the browser.
2. Access the integrations page with both.
3. In the first tab, click + and enter the host.
4. In the second tab, click + and enter the same host.

Reply:
Ok.